### PR TITLE
Drop `git ls-files` in gemspec

### PR DESCRIPTION
**Pull request by utkarsh2102**
Hi again 👋🏻 

As discussed in #1353, this PR drops `git` in gemspec, thereby also helping in smooth downstream maintenance (in Debian, Fedora, and so on).
Also adds the `Packaging` extension of RuboCop. More about it can be found on https://docs.rubocop.org/rubocop-packaging.

Closes: #1353 

Signed-off-by: Utkarsh Gupta <<utkarsh@debian.org>>

### CODE REVIEW

File: tasks/documentation.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require_relative '../lib/shoulda/matchers/version'
+require 'shoulda/matchers/version'
```
Comment (KapilSachdev) [inline tasks/documentation.rb:1]: Why this change is done? Seems unrelated to the PR.

File: Gemfile [hunk 2]
```diff
@@ -7,6 +7,7 @@ gem 'pry-byebug'
 gem 'rake', '13.0.1'
 gem 'rspec', '~> 3.9'
 gem 'rubocop', require: false
+gem 'rubocop-packaging', require: false
```
Comment (KapilSachdev) [inline Gemfile:10]: Though it misses a `gemlock` entry but why this has been added?
Has there been any reported issues or problems that this addition helps in resolving ?

File: tasks/documentation.rb (hunk 1 above)
Comment (utkarsh2102) [reply tasks/documentation.rb:1]: Each change has been committed separately with a verbose commit message.
Please let me know if they're not clear enough.

File: Gemfile (hunk 2 above)
Comment (utkarsh2102) [reply Gemfile:10]: `Gemfile.lock` has now been updated, thanks!
It helps us in maintaining downstream (links in the commit message :)). More about these cops can be found at https://docs.rubocop.org/rubocop-packaging and this has a style guide hosted at https://packaging.rubystyle.guide.

Comment (KapilSachdev) [reply Gemfile:10]: Ok, so `require_relative` was part of this, got it :+1:

File: tasks/documentation.rb (hunk 1 above)
Comment (VSPPedro) [reply tasks/documentation.rb:1]: Nice commit message. I did not know that. Thanks! :+1: 

I'm just wondering, the project has 86 uses of require_relative, so why modify just this one?

If we intend to change several require_relative to require, I think it would be better to separate into another PR, but for now, I think we need to wait for more feedback on this change.

Comment (utkarsh2102) [reply tasks/documentation.rb:1]: Hehe, nice! 😄 
The rationale for each cop is documented here: https://docs.rubocop.org/rubocop-packaging/cops_packaging.html#require-relative-hardcoding-lib-rationale

So it's still preferred to use `require_relative` but just not cross-lib. More specifically, defintiely not from the test code to `lib/`.

**Comment by utkarsh2102** [comment]
@mcmire, should be ready for your review now, could you please take a look? 😄

**Comment by mcmire** [comment]
Thank you for these changes! Merged!
